File: 121.BestTimetoBuyAndSellStock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0;
        int min_price = prices[0];
        for (auto &price:prices)
        {
            int profit = price - min_price;
            max_profit = max(profit,max_profit);
            min_price = min(price,min_price);
        }
        return max_profit;
    }
};

File: 121.BestTimetoBuyAndSellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        min_price = prices[0]
        for price in prices:
            profit = price - min_price
            max_profit = max(profit,max_profit)
            min_price = min(min_price,price)
        return max_profit